[feat] 지역(동네) 목록 조회 기능

pie2457 opened the pull request:
## Issues
- #42 

## What is this PR? 👓
지역(동네) 목록 조회 기능에 관한 PR입니다.

## Key changes 🔑
- 지역(동네) 목록 조회 
- 지역(동네) 목록 조회 테스트

pie2457 committed:
feat #42 : 지역(동네) 목록 조회 기능

pie2457 committed:
test #42 : 지역(동네) 목록 조회 기능 테스트

pie2457 committed:
style #42 : 사용되지 않는 괄호 삭제

23Yong commented on src/main/java/pie/tomato/tomatomarket/infrastructure/persistence/region/RegionRepository.java:
```diff
@@ -1,8 +1,28 @@
+			return null;
+		}
+
+		return region.fullAddressName.like("%" + addressName + "%");
```
like 연산에 양 옆으로 `%`가 붙어 있으면 인덱스를 타지 않는다는 사실~

그런데 어차피 region 테이블에는 데이터가 그렇게 많지 않으니까(만개도 안되지 않나요?) 풀테이블스캔이 더 빠를 수 있을 것 같네요

pie2457 replied:
몰랐네요ㅠㅠ 감사합니다 수정해볼게요 !

pie2457 replied:
만개 .. ㅎㅎ 천개도 안된답니다 풀테이블 스캔으로 가야겠어요 like만 contains로 바꿔야겠네여 !

23Yong commented on src/main/java/pie/tomato/tomatomarket/application/memberTown/MemberTownService.java:
```diff
@@ -82,4 +87,14 @@ private MemberTown getMemberTown(SelectMemberTownRequest selectMemberTownRequest
+			size, cursor);
+		List<MemberTownListResponse> content = responses.getContent();
+
+		Long nextCursor = content.isEmpty() ? null : content.get(content.size() - 1).getAddressId();
```
이 부분 계속 반복되고 있는 것 같은데 좋은 방법이 없는지 고민되네요

pie2457 replied:
저도 같은 고민을 매번하지만 딱히 방법을 찾지 못하고 있네요 ㅠㅡㅠ..

23Yong reviewed (approved):
수고하셨어요~!

pie2457 committed:
refactor #42 : 리뷰사항 반영

- MemberTownService : 중복 로직 메서드로 분리
- RegionRepository : like -> contains로 변경
앞 뒤로 붙은 퍼센트의 역할을 contains가 함
